### PR TITLE
add stamped screenshot preview

### PR DESCRIPTION
**Pull request by poludnev**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -58,12 +63,88 @@ function App() {
...
+          );
+          ctx.drawImage(
+            stamp,
+            offset.x,
+            offset.y,
+            stampFitCanvasSizes.width,
+            stampFitCanvasSizes.height,
+          );
+
+          ctx.font = '5px monospace';
```
Comment (utgarda) [inline src/App.tsx:125]: Can this be a top-level constant or a config value?

File: src/App.tsx [hunk 2]
```diff
@@ -58,12 +63,88 @@ function App() {
...
+          ctx.fillStyle = 'red';
+          const ctxFillTextX = 10;
+          const ctxFillTextY = 5;
+
+          const metaPreview = getPreviewMetadata(requestUrl);
+
+          metaPreview.forEach((string, index) => {
+            const stringWidth = ctx.measureText(string).width;
+            if (stringWidth > 480) {
```
Comment (utgarda) [inline src/App.tsx:134]: magic number

File: src/utils/index.ts [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+    expires: date,
+    'last-modified': date,
+    location: 'https://ya.ru/?nr=1&redirect_ts=1676310742.38901',
+    nel: '{"report_to": "network-errors", "max_age": 100, "success_fraction": 0.001, "failure_fraction": 0.1}',
+    p3p: 'policyref="/w3c/p3p.xml", CP="NON DSP ADM DEV PSD IVDo OUR IND STP PHY PRE NAV UNI"',
+    'x-content-type-options': 'nosniff',
+    'x-req-id':
+      '1676310742355222-10222453901922640994-sas2-0311-sas-l7-balancer-8080-BAL-9322',
+  };
+  // const url = new URL(sourceUrl);
```
Comment (utgarda) [inline src/utils/index.ts:17]: .

File: src/utils/index.ts [hunk 4]
```diff
@@ -0,0 +1,91 @@
+export const getPreviewMetadata = (url: string, ip: string = '0.0.0.0') => {
+  const date = new Date(Date.now()).toString();
+
+  const headers: { [id: string]: string } = {
+    'accept-ch':
```
Comment (utgarda) [inline src/utils/index.ts:5]: Please add a comment explaining how and this bogus metadata never gets to the final result.

File: src/utils/index.ts [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+  width: number;
+} => {
+  const srcRatio = srcWidth / srcHeight;
+  if ((srcRatio < 1 || canvasWidth >= canvasHeight) && canvasHeight * srcRatio < canvasWidth) {
+    return { height: canvasHeight, width: canvasHeight * srcRatio };
+  }
+  return { height: canvasWidth / srcRatio, width: canvasWidth };
+};
+
+export const getOffset = (
```
Comment (utgarda) [inline src/utils/index.ts:66]: Please add usage contract - say that equal sizes aren't supposed to be passed as args.

File: src/utils/index.ts [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+  offsetYRatio: number = 0,
+): { x: number; y: number } => {
+  if (imageWidth > canvasWidth || imageHeight > canvasHeight)
+    throw new Error('image bigger that canvas');
+
+  if (
+    (imageWidth === canvasWidth && offsetXRatio !== 0) ||
+    (imageHeight === canvasHeight && offsetYRatio !== 0)
+  )
+    throw new Error('cannot offset if equal sizes');
```
Comment (utgarda) [inline src/utils/index.ts:81]: [typescript assert](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#assertion-functions)

File: src/utils/index.ts [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+
+  if (Math.abs(offsetXRatio) > 1 || Math.abs(offsetYRatio) > 1)
+    throw new Error('offsetRaion more than 1');
```
Comment (utgarda) [inline src/utils/index.ts:84]: .

**Review by utgarda — CHANGES_REQUESTED** [review]
Please take care of the cosmetic issues and 

manage to run the tests in Github Actions
https://github.com/rainbow-me/rainbowkit/issues/461
@Elizavetta please help with the above

**Review by Elizavetta — APPROVED** [review]
tests are running; minor improvements done;

**Pull request merged**